contrib/buildsystems/CMakeLists.txt

```diff
@@ -63,10 +63,20 @@ endif()
 
 if(NOT DEFINED CMAKE_EXPORT_COMPILE_COMMANDS)
 	set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
+	message("settting CMAKE_EXPORT_COMPILE_COMMANDS: ${CMAKE_EXPORT_COMPILE_COMMANDS}")
 endif()
 
 if(USE_VCPKG)
 	set(VCPKG_DIR "${CMAKE_SOURCE_DIR}/compat/vcbuild/vcpkg")
+	message("WIN32: ${WIN32}") # show its underlying text values
+	message("VCPKG_DIR: ${VCPKG_DIR}")
+	message("VCPKG_ARCH: ${VCPKG_ARCH}") # maybe unset
+	message("MSVC: ${MSVC}")
+	message("CMAKE_GENERATOR: ${CMAKE_GENERATOR}")
+	message("CMAKE_CXX_COMPILER_ID: ${CMAKE_CXX_COMPILER_ID}")
+	message("CMAKE_GENERATOR_PLATFORM: ${CMAKE_GENERATOR_PLATFORM}")
+	message("CMAKE_EXPORT_COMPILE_COMMANDS: ${CMAKE_EXPORT_COMPILE_COMMANDS}")
+	message("ENV(CMAKE_EXPORT_COMPILE_COMMANDS): $ENV{CMAKE_EXPORT_COMPILE_COMMANDS}")
```

Comment:
Good. Please make sure to capitalize `CMake` correctly in the commit message. It has never been, and will never be, `Cmake`.

Reply:
The `${VCPKG_ARCH})` check is still essential because when the CMakeLists.txt auto detection by Visual Studio happens CMake (VS internal) will run the CMakeLists.txt without any parameters, so `${VCPKG_ARCH})` won't be set and the script fails. 

We need to use the default value that will have been set by the `vcpkg_install.bat` which may have also been run by the user without parameters. It is a backward compatibility requirement.

I've also corrected the CMake spelling in the commit message.

Reply:
Also the `message("VCPKG_ARCH: unset, using 'x64-windows'")`, for me, is part of providing clarity of the build-options information.